packages/ra-ui-materialui/src/layout/CheckForApplicationUpdate.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import { ReactElement } from 'react';
+import {
+    useNotify,
+    UseCheckForApplicationUpdateOptions,
+    useCheckForApplicationUpdate,
+} from 'ra-core';
+import { ApplicationUpdatedNotification } from './ApplicationUpdatedNotification';
+
+/**
+ * A component which checks for a new version of the application and can display a notification asking users to reload the page.
+ *
+ * @param {CheckForApplicationUpdateProps} props
+ * @param {boolean} options.disabled Optional. Whether the check should be disabled. Defaults to false.
+ * @param {string|ReactElement} props.notification The notification to display to the user. Displayed only if `updateMode` is manual. Defaults to `<ApplicationUpdatedNotification />`.
+ * @param {string} options.url Optional. The URL to download to check for code update. Defaults to the current URL.
+ * @param {number} options.checkInterval Optional. The interval in milliseconds between two checks. Defaults to 3600000 (1 hour).
+ *
+ * @example <caption>Basic usage</caption>
+ * import { Admin, Resource, Layout, CheckForApplicationUpdate, ListGuesser } from 'react-admin';
+ *
+ * const MyLayout = ({ children, ...props }) => (
+ *   <Layout {...props}>
+ *     {children}
+ *     <CheckForApplicationUpdate />
+ *   </Layout>
+ * );
+ *
+ * const App = () => (
+ *   <Admin layout={MyLayout}>
+ *      <Resource name="posts" list={ListGuesser} />
+ *   </Admin>
+ * );
+ *
+ * @example <caption>Custom notification</caption>
+ * import { forwardRef } from 'react';
+ * import { Admin, Resource, Layout, CheckForApplicationUpdate, ListGuesser } from 'react-admin';
+ *
+ * const CustomAppUpdatedNotification = forwardRef((props, ref) => (
+ *   <Alert
+ *     ref={ref}
+ *     severity="info"
+ *     action={
+ *       <Button
+ *         color="inherit"
+ *         size="small"
+ *         onClick={() => window.location.reload()}
+ *       >
+ *         Update
+ *       </Button>
+ *     }
+ *   >
+ *     A new version of the application is available. Please update.
+ *   </Alert>
+ * ));
+ *
+ * const MyLayout = ({ children, ...props }) => (
+ *   <Layout {...props}>
+ *     {children}
+ *     <CheckForApplicationUpdate notification={<CustomAppUpdatedNotification />} />
+ *   </Layout>
+ * );
+ *
+ * const App = () => (
+ *   <Admin layout={MyLayout}>
+ *      <Resource name="posts" list={ListGuesser} />
+ *   </Admin>
+ * );
+ */
+export const CheckForApplicationUpdate = (
+    props: CheckForApplicationUpdateProps
+) => {
+    const { notification = DEFAULT_NOTIFICATION, ...rest } = props;
+    const notify = useNotify();
+
+    const onNewVersionAvailable = () => {
+        notify(notification, {
+            type: 'info',
+            autoHideDuration: 0,
```

Comment:
What happens if the user wants to dismiss this notification (e.g. they're in the middle of an update)?

Reply:
Now we store the last version hash. If users dismiss the notification, they won't see it again unless a newer version is released.